[TRTLLM-7958][doc] add 1.0 release notes

nv-guomingz opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Release 1.0: broader model support (large multimodal and MoE models), LoRA and PyTorch backend enhancements, improved KV-cache, streaming/dynamic decoding, FP8 and other quantization improvements, attention/MLA scheduling, disaggregation, expanded benchmarking.

- Infrastructure
  - Upgraded base images and core dependencies for improved stability and performance.

- Breaking Changes / API
  - PyTorch promoted as the default backend; runtime defaults and public configuration shapes changed; new performance/quantization options exposed.

- Bug Fixes
  - Fixes for MLA, disaggregated serving, scheduling, memory, and workflow issues.

- Known Issues
  - Platform-specific disaggregation and multi-node caveats remain.

- Documentation
  - Added comprehensive Release 1.0 notes and updated API docs.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

<!--
Please write the PR title by following this template:

**[JIRA ticket/NVBugs ID/GitHub issue/None][type] Summary**

Valid ticket formats:
  - JIRA ticket: [TRTLLM-1234] or [FOOBAR-123] for other FOOBAR project
  - NVBugs ID: [https://nvbugs/1234567]
  - GitHub issue: [#1234]
  - No ticket: [None]

Valid types (lowercase): [fix], [feat], [doc], [infra], [chore], etc.

Examples:
  - [TRTLLM-1234][feat] Add new feature
  - [https://nvbugs/1234567][fix] Fix some bugs
  - [#1234][doc] Update documentation
  - [None][chore] Minor clean-up

Alternative (faster) way using CodeRabbit AI:

**[JIRA ticket/NVBugs ID/GitHub issue/None] @coderabbitai title**

NOTE: "@coderabbitai title" will be replaced by the title generated by CodeRabbit AI, that includes the "[type]" and title.
For more info, see /.coderabbit.yaml.

-->

## Description

<!--
Please explain the issue and the solution in short.
-->

## Test Coverage

<!--
Please list clearly what are the relevant test(s) that can safeguard the changes in the PR. This helps us to ensure we have sufficient test coverage for the PR.
-->

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- The reviewers assigned automatically/manually are appropriate for the PR.


- [ ] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--reuse-test (optional)pipeline-id --disable-fail-fast --skip-test --stage-list "A10-PyTorch-1, xxx" --gpu-type "A30, H100_PCIe" --test-backend "pytorch, cpp" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx" --detailed-log --debug(experimental)]`

Launch build/test pipelines. All previously running jobs will be killed.

`--reuse-test (optional)pipeline-id ` *(OPTIONAL)* : Allow the new pipeline to reuse build artifacts and skip successful test stages from a specified pipeline or the last pipeline if no pipeline-id is indicated. If the Git commit ID has changed, this option will be always ignored. The DEFAULT behavior of the bot is to reuse build artifacts and successful test results from the last pipeline.

`--disable-reuse-test ` *(OPTIONAL)* : Explicitly prevent the pipeline from reusing build artifacts and skipping successful test stages from a previous pipeline. Ensure that all builds and tests are run regardless of previous successes.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-PyTorch-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-PyTorch-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--test-backend "pytorch, cpp"` *(OPTIONAL)* : Skip test stages which don't match the specified backends. Only support [pytorch, cpp, tensorrt, triton]. Examples: "pytorch, cpp" (does not run test stages with tensorrt or triton backend). Note: Does **NOT** update GitHub pipeline status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests in addition to running L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx".

`--detailed-log ` *(OPTIONAL)* : Enable flushing out all logs to the Jenkins console. This will significantly increase the log volume and may slow down the job.

`--debug ` *(OPTIONAL)* : **Experimental feature**. Enable access to the CI container for debugging purpose. Note: Specify exactly one stage in the `stage-list` parameter to access the appropriate container environment. Note: Does **NOT** update GitHub check status.

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`
and the `scripts/test_to_stage_mapping.py` helper.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

nv-guomingz committed:
[TRTLLM-7958][doc] add 1.0 release notes

Signed-off-by: nv-guomingz <137257613+nv-guomingz@users.noreply.github.com>

amitz-nv commented on docs/source/release-notes.md:
```diff
@@ -4,6 +4,173 @@
+
+
+### Fixed Issues
+- Fix illeagel memory access in MLA (#6437)
```
Typo: illeagel -> illegal

nv-guomingz replied:
Thanks, Fixed.

nv-guomingz commented on docs/source/release-notes.md:
```diff
@@ -4,6 +4,173 @@
+- The dependent xgrammar version is updated to 0.1.21.
+- The dependent transformers version is updated to 4.51.3.
+
+### API Changes
```
Could you please provide a comprehensive description of #6220? I believe the original PR's title is somewhat generic.

nv-guomingz replied:
Updated and I refined the first statement base on your input.

pcastonguay commented on docs/source/release-notes.md:
```diff
@@ -4,6 +4,173 @@
+- Fix the unexpected keyword argument 'streaming' (#5436)
+
+### Known Issues
+- On bare-metal Ubuntu 22.04 or 24.04, please install the `cuda-python==12.9.1` package after installing the TensorRT-LLM wheel. This resolves an incompatibility issue with the default cuda-python 13 of error `ImportError: cannot import name 'cuda' from 'cuda'`.
```
```suggestion
- On bare-metal Ubuntu 22.04 or 24.04, please install the `cuda-python==12.9.1` package after installing the TensorRT-LLM wheel. This resolves an incompatibility issue with the default cuda-python 13 of error `ImportError: cannot import name 'cuda' from 'cuda'`.
- When using disaggregated serving with pipeline parallelism and KV cache reuse, a hang can occur. This will be fixed in a future release. In the meantime, disabling KV cache reuse will fix this issue. 
```

schetlur-nv replied:
@pcastonguay can this happen with PP=2 too? I _thought_ there was something about this only happening with PP=4?

pcastonguay committed:
Include release note for disagg + reuse + pp known issue

Signed-off-by: pcastonguay <55748270+pcastonguay@users.noreply.github.com>

laikhtewari reviewed (commented):
@nv-guomingz @juney-nvidia where do we want to talk about LLM API stability?

schetlur-nv committed:
Update release-notes.md

Add relnote for multi-node single GPU per node exception

Signed-off-by: Sharan Chetlur <116769508+schetlur-nv@users.noreply.github.com>

schetlur-nv commented on docs/source/release-notes.md:
```diff
@@ -4,6 +4,177 @@
+### Known Issues
+- On bare-metal Ubuntu 22.04 or 24.04, please install the `cuda-python==12.9.1` package after installing the TensorRT-LLM wheel. This resolves an incompatibility issue with the default cuda-python 13 of error `ImportError: cannot import name 'cuda' from 'cuda'`.
+- When using disaggregated serving with pipeline parallelism and KV cache reuse, a hang can occur. This will be fixed in a future release. In the meantime, disabling KV cache reuse will fix this issue.
+- Running multi-node cases where each node has just a single GPU is known to fail. This will be addressed in a future release. 
```
@SimengLiu-nv for viz.

nv-guomingz committed:
add documentation section.

Signed-off-by: nv-guomingz <137257613+nv-guomingz@users.noreply.github.com>

nv-guomingz commented:
> > @nv-guomingz @juney-nvidia where do we want to talk about LLM API stability?
> 
> * @Superjomn for vis

Updated: I've added additional section for documentation changes

https://github.com/NVIDIA/TensorRT-LLM/pull/7605/commits/9026f574d0d7a82f9607888fdd38d5d5f5a3f367#diff-9bc5639e30bf7e6d6cc971c19e3294c3803443eefa5f3a27d33e313ee5bc68e5R95

schetlur-nv commented on docs/source/release-notes.md:
```diff
@@ -4,6 +4,181 @@
+
+### Known Issues
+- On bare-metal Ubuntu 22.04 or 24.04, please install the `cuda-python==12.9.1` package after installing the TensorRT-LLM wheel. This resolves an incompatibility issue with the default cuda-python 13 of error `ImportError: cannot import name 'cuda' from 'cuda'`.
+- When using disaggregated serving with pipeline parallelism and KV cache reuse, a hang can occur. This will be fixed in a future release. In the meantime, disabling KV cache reuse will fix this issue.
```
What's the motivation to lift this restriction? Do we have planned work? If it's something that will not be fixed imminently, we should move this somewhere else in the documentation and not the release notes (which should be release specific).
FWIW, moving it could be done in the next release, doesn't have to be in 1.0.

laikhtewari reviewed (changes requested):
I suggest including the following intro before going into the detailed release notes

"TensorRT LLM 1.0 brings 2 major changes: the PyTorch-based architecture is now stable and the default experience, and the LLM API is now stable. For more details on new developments in 1.0, please see below"

Currently, we're burying the lede on PyTorch and LLM API

nv-guomingz commented on docs/source/release-notes.md:
```diff
@@ -4,6 +4,181 @@
+- Fix the unexpected keyword argument 'streaming' (#5436)
+
+### Known Issues
+- On bare-metal Ubuntu 22.04 or 24.04, please install the `cuda-python==12.9.1` package after installing the TensorRT-LLM wheel. This resolves an incompatibility issue with the default cuda-python 13 of error `ImportError: cannot import name 'cuda' from 'cuda'`.
```
Got it. Let me update the known issues section.

nv-guomingz commented on docs/source/release-notes.md:
```diff
@@ -4,6 +4,181 @@
+- The dependent public PyTorch version is updated to 2.8.0.
+- The dependent NVIDIA ModelOpt version is updated to 0.33.
+- The dependent xgrammar version is updated to 0.1.21.
+- The dependent transformers version is updated to 4.51.3.
```
Fixed

nv-guomingz commented on docs/source/release-notes.md:
```diff
@@ -4,6 +4,181 @@
+### Infrastructure Changes
+- The base Docker image for TensorRT-LLM is updated to `nvcr.io/nvidia/pytorch:25.06-py3`.
+- The base Docker image for TensorRT-LLM Backend is updated to `nvcr.io/nvidia/tritonserver:25.06-py3`.
+- The dependent public PyTorch version is updated to 2.8.0.
```
Fixed

nv-guomingz commented:
> I suggest including the following intro before going into the detailed release notes
> 
> "TensorRT LLM 1.0 brings 2 major changes: the PyTorch-based architecture is now stable and the default experience, and the LLM API is now stable. For more details on new developments in 1.0, please see below"
> 
> Currently, we're burying the lede on PyTorch and LLM API

Done

nv-guomingz commented:
/bot skip --comment "docs only change"

tensorrt-cicd commented:
[PR_Github #18875](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/18875/) [ skip ] completed with state `SUCCESS`
Skipping testing for commit `6673690`

tensorrt-cicd commented:
[PR_Github #18908](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/18908/) [ skip ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #18908](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/18908/) [ skip ] completed with state `SUCCESS`
Skipping testing for commit `c5d0c92`
